### PR TITLE
Store scabbard transaction receipts

### PR DESCRIPTION
**Pull request by ltseeley**
Adds the `TransactionReceiptStore` from the `sawtooth` crate to
`ScabbardState` for storing transaction receipts that are committed. The
receipt store provides an absolute history of all transactions that were
committed by the scabbard service, which can be used to "catch-up"
state-delta subscribers that may be behind.

Signed-off-by: Logan Seeley <seeley@bitwise.io>

### CODE REVIEW

File: libsplinter/src/service/scabbard/factory.rs [hunk 1]
```diff
@@ -30,26 +30,34 @@ use crate::signing::SignatureVerifierFactory;
 
 use super::{Scabbard, SERVICE_TYPE};
 
-const DEFAULT_DB_DIR: &str = "/var/lib/splinter";
-const DEFAULT_DB_SIZE: usize = 1028 * 1028 * 1028;
+const DEFAULT_STATE_DB_DIR: &str = "/var/lib/splinter";
+const DEFAULT_STATE_DB_SIZE: usize = 1028 * 1028 * 1028;
```
Comment (jsmitchell) [inline libsplinter/src/service/scabbard/factory.rs]: Why are these 1028? (I realize this predates this commit)

Comment (ltseeley) [reply libsplinter/src/service/scabbard/factory.rs]: Just the precedent set by Sawtooth.

Comment (ltseeley) [reply libsplinter/src/service/scabbard/factory.rs]: Uhhh yeah that should actually be 1024. No idea why it's 1028.

Comment (ltseeley) [reply libsplinter/src/service/scabbard/factory.rs]: Fixed

File: splinterd/src/daemon.rs [hunk 2]
```diff
@@ -289,6 +289,8 @@ impl SplinterDaemon {
                 })?;
         let orchestrator = ServiceOrchestrator::new(
             vec![Box::new(ScabbardFactory::new(
+                None,
```
Comment (peterschwarz) [inline splinterd/src/daemon.rs]: These options feel like they should probably be a single config object: reading this is pretty heavy on the `None`'s

**Pull request merged**